### PR TITLE
Include documentation

### PR DESCRIPTION
**Pull request by local-minimum**
* The `rac -help` still gives a short info about how to run the program
* The `rac -help output` gives extensive information about the output data
* Then there's `rac -help CCD` and the other output streams

I opted for a very long text string because it seemed simplest way to build the info into the binary.

I only did the CCD.csv now because there's a lot of text there already. If you see typos or things that need better explanation, feel very free to use the suggestion feature in the review.

Manually tested.

### CODE REVIEW

File: cmd/rac/output.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+- "RAMES": Means the following column says what RAMSES specification was used.
+- "SPU045-S2:6F": The RAMSES version
+- "INNOSAT": Means the following column says what INNOSAT specification was
+  used.
+- "IS-OSE-ICD-0005:1": The INNOSAT version
+- "AEZ": Means the following column says what AEZ specification was used.
+- "AEZICD002:H": The AEZ version
+
+The header row starts with a couple of columns common for all output and then
+follows columns specific to each file
```
Comment (skymandr) [inline cmd/rac/output.go]:
```suggestion
The header row starts with a couple of columns common to all output and then
follows columns specific to each file

```

File: cmd/rac/output.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+The header row starts with a couple of columns common to all output and then
+follows columns specific to each file
+
+
+- File
+  The full path to the rac-file on the computer that produced the csv
+- ProcessingDate
+  The local time of when the file was processed
+- RamsesTime (Ramses Header)
+  The time of when the ramses file was created (UTC)
```
Comment (skymandr) [inline cmd/rac/output.go]:
```suggestion
  The local time when the file was processed
- RamsesTime (Ramses Header)
  The time when the ramses file was created (UTC)

```

File: cmd/rac/output.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+- RamsesTime (Ramses Header)
+  The time when the ramses file was created (UTC)
+
+- QualityIndicator (Ramses TM Header)
+  Indicates whether the transported data is complete or partial
+  (0 = Complete, 1 = partial).
+- LossFlag (Ramses TM Header)
+  Used to indicate that a sequence discontinuity has been detected
+- VCFrameCounter (Ramses TM Header)
+  Counter of the transfer fram the payload packet arrived in.
```
Comment (skymandr) [inline cmd/rac/output.go]:
```suggestion
  Counter of the transfer frame the payload packet arrived in.

```

File: cmd/rac/output.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+- QualityIndicator (Ramses TM Header)
+  Indicates whether the transported data is complete or partial
+  (0 = Complete, 1 = partial).
+- LossFlag (Ramses TM Header)
+  Used to indicate that a sequence discontinuity has been detected
+- VCFrameCounter (Ramses TM Header)
+  Counter of the transfer frame the payload packet arrived in.
+
+- SPSequenceCount (Innosat Source Header)
```
Comment (skymandr) [inline cmd/rac/output.go:38]: No description?

Comment (skymandr) [inline cmd/rac/output.go:40]: No description?

File: cmd/rac/output.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+- TMHeaderNanoseconds
+  The time of the TM packet creation (nanoseconds since epoch)
+- SID
+  The name of the SID or empty if the packet has no SID
+- RID
+  The name of the RID or empty if the packet has no RID
+
+Note that each line should have a SID or a RID depending on packet type
+
+All files also has a final common column
```
Comment (skymandr) [inline cmd/rac/output.go]:
```suggestion
All files also have a final common column

```

File: cmd/rac/output.go [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  The name of the SID or empty if the packet has no SID
+- RID
+  The name of the RID or empty if the packet has no RID
+
+Note that each line should have a SID or a RID depending on packet type
+
+All files also have a final common column
+
+- Error
+  If an error occurred it will be written here.
```
Comment (skymandr) [inline cmd/rac/output.go:59]: Can be a list of errors, right?

File: cmd/rac/output.go [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  If empty, then no error occurred extracting the data.
+
+### CCD.csv ###
+The following columns directly export the data in the rac:
+CCDSEL, WDWOV, JPEGQ, FRAME, NROW, NRBIN, NRSKIP, NCOL, NCSKIP, NFLUSH
+TEXPMS, TEMP, FBINOV, LBLNK, TBLNK, ZERO, TIMING1, TIMING2, VERSION
+TIMING3, NBC, BC
+
+The following columns parses the values further:
```
Comment (skymandr) [inline cmd/rac/output.go]:
```suggestion
The following columns parse the values further:
```

File: cmd/rac/output.go [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+- WDW InputDataWindow
+  Written as the from - to bits used in the original image
+  "11..0" (value in rac 0x0)
+  "12..1" (value in rac 0x1)
+  "13..2" (value in rac 0x2)
+  "14..3" (value in rac 0x3)
+  "15..4" (value in rac 0x4)
+  "15..0" which is the full image (value in rac 0x7)
+- NCBIN FPGAColumns
+  The actual number of FPGA Columns (value in rac is the exponent in 2^x)
```
Comment (skymandr) [inline cmd/rac/output.go]:
```suggestion
  The actual number of FPGA Columns (value in rac is the exponent in 2^N)

```

File: cmd/rac/output.go [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+  Written as the from - to bits used in the original image
+  "11..0" (value in rac 0x0)
+  "12..1" (value in rac 0x1)
+  "13..2" (value in rac 0x2)
+  "14..3" (value in rac 0x3)
+  "15..4" (value in rac 0x4)
+  "15..0" which is the full image (value in rac 0x7)
+- NCBIN FPGAColumns
+  The actual number of FPGA Columns (value in rac is the exponent in 2^N)
+
```
Comment (skymandr) [inline cmd/rac/output.go]: Is this blank line intentional?

File: cmd/rac/output.go [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+- NCBIN CCDColumns
+  The number of CCD Columns
+- GAIN Mode
+  "High" (value in rac 0b0)
+  "Low" (value in rac 0b1)
+- GAIN Timing
+  "Faster" (value in rac 0b0)
```
Comment (skymandr) [inline cmd/rac/output.go]: I realise this is probably what the spec says, but I can't help but think that it isn't helpful. I mean, which is the fastest? "Full" or "Faster"...?

File: cmd/rac/output.go (hunk 6 above)
Comment (local-minimum) [reply cmd/rac/output.go:59]: No the DataRecord object only holds one error so I don't see how that would happen.

File: cmd/rac/output.go (hunk 9 above)
Comment (local-minimum) [reply cmd/rac/output.go]: Nope, I was removing untreated lines to above and accidentally left this one.

Comment (local-minimum) [inline cmd/rac/output.go]:
```suggestion

```

File: cmd/rac/output.go (hunk 4 above)
Comment (local-minimum) [reply cmd/rac/output.go:40]: duplicated comment?

File: cmd/rac/output.go [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+- QualityIndicator (Ramses TM Header)
+  Indicates whether the transported data is complete or partial
+  (0 = Complete, 1 = partial).
+- LossFlag (Ramses TM Header)
+  Used to indicate that a sequence discontinuity has been detected
+- VCFrameCounter (Ramses TM Header)
+  Counter of the transfer frame the payload packet arrived in.
+
+- SPSequenceCount (Innosat Source Header)
```
Comment (local-minimum) [inline cmd/rac/output.go:40]:
```suggestion
- SPSequenceCount (Innosat Source Header)
  A counter that increases with each packet, may never short cycle and should wrap around to zero after 2^14-1

```

Comment (local-minimum) [reply cmd/rac/output.go:40]: @joakimmoller is that the correct counter description?

File: cmd/rac/output.go [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+  The actual number of FPGA Columns (value in rac is the exponent in 2^N)
+
+- NCBIN CCDColumns
+  The number of CCD Columns
+- GAIN Mode
+  "High" (value in rac 0b0)
+  "Low" (value in rac 0b1)
+- GAIN Timing
+  "Faster" (value in rac 0b0)
+  "Full" (value in rac 0b1)
```
Comment (local-minimum) [inline cmd/rac/output.go]:
```suggestion
  "Faster" used for binned and discarded (value in rac 0b0)
  "Full" used even for pixels that are not read out (value in rac 0b1)
```

Comment (local-minimum) [reply cmd/rac/output.go]: I suppose it's gain that is used :shrug:

File: cmd/rac/output.go (hunk 4 above)
Comment (skymandr) [reply cmd/rac/output.go:40]: Yes. No description?

File: cmd/rac/output.go (hunk 11 above)
Comment (skymandr) [reply cmd/rac/output.go:40]: `s/zeron/zero/`

File: cmd/rac/output.go (hunk 6 above)
Comment (skymandr) [reply cmd/rac/output.go:59]: Right, yes. CCD only. Will you make an issue for the remainder?

File: cmd/rac/output.go (hunk 4 above)
Comment (local-minimum) [reply cmd/rac/output.go:40]: Added suggestion, but that spec is so confusing to read to just wanted to check with joakim before committing it.

File: cmd/rac/output.go (hunk 9 above)
Comment (local-minimum) [reply cmd/rac/output.go]: added suggestion, will commit in batch

File: cmd/rac/output.go (hunk 10 above)
Comment (local-minimum) [reply cmd/rac/output.go]: added the text that exists in the specification as a suggestion

File: cmd/rac/output.go (hunk 6 above)
Comment (local-minimum) [reply cmd/rac/output.go:59]: The error is for any packet, we are confuse

File: cmd/rac/output.go (hunk 11 above)
Comment (joakimmoller) [reply cmd/rac/output.go:40]: Correct description of SPSSequenceCount. You could also mention that the VCFrameCounter wraps at (2^16)-1 i.e. (modulo 65536)

File: cmd/rac/output.go [hunk 13]
```diff
@@ -0,0 +1,263 @@
...
+  (0 = Complete, 1 = partial).
+- LossFlag (Ramses TM Header)
+  Used to indicate that a sequence discontinuity has been detected
+- VCFrameCounter (Ramses TM Header)
+  Counter of the transfer frame the payload packet arrived in.
+  Wraps at (2^16)-1 i.e.
+
+- SPSequenceCount (Innosat Source Header)
+  A counter that increases with each packet, may never short cycle and should
+  wrap around to zeron after 2^14-1
```
Comment (local-minimum) [inline cmd/rac/output.go]:
```suggestion
  wrap around to zero after 2^14-1

```

File: cmd/rac/output.go [hunk 14]
```diff
@@ -0,0 +1,263 @@
...
+  The time when the ramses file was created (UTC)
+
+- QualityIndicator (Ramses TM Header)
+  Indicates whether the transported data is complete or partial
+  (0 = Complete, 1 = partial).
+- LossFlag (Ramses TM Header)
+  Used to indicate that a sequence discontinuity has been detected
+- VCFrameCounter (Ramses TM Header)
+  Counter of the transfer frame the payload packet arrived in.
+  Wraps at (2^16)-1 i.e.
```
Comment (local-minimum) [inline cmd/rac/output.go]:
```suggestion
  Wraps at (2^16)-1

```

File: cmd/rac/output.go [hunk 15]
```diff
@@ -0,0 +1,265 @@
...
+.   .      .    *     .     .    .      .   .       .  .                     .
+    .           .           .           .           .         +  .
+. .        .  .       .   .      .    .     .     .    .      .   .    +
+
+.   +      .          ___/\_._/~~\_...__/\__.._._/~\        .         .   .
+ .               _.--'                              '--./\          .   .
+          /~~\/~\                                         '-/~\_            .
+.      .-'                                                      '-/\_
+_/\.-'                                                               \_/~\/\-.__
+                                                                                '.
```
Comment (local-minimum) [inline cmd/rac/output.go]:
```suggestion
+--------------------------------------------------------------------------------+
|..                 .                                             ..    .        |
|    .                              .                                   ..       |
|                                        .   .             ..            .       |
|..                                                         .        . ~.        |
| .                            ..    .       ..                                  |
| .                                       ~~:~::~~.         .                    |
|                                     ~~~~~~~~~~:+++~~..                        .|
|                                ..~~~~~~~.~~~~~~~~:+++:~.   .                   |
|                             .~~:~~~~.~~~~~~~~~~::::::+oo+.                     |
|                          .~~~~~.~.~~~~~~~~~~~::::::+o++:~.                     |
|                          ~o+:~~~~~~~~~~~~~::::::++++~.                         |
|..                        .~.~:++:~~~~~::::::+:o+:~~:.                          |
|o+o++:~~~..               .~   ~:+++++::::++:~.+::+~..~~.                       |
|~::::++++++::~~~...       .~ ~~~~~:::+++++~.   ~:+~~~~..~:.                     |
|       ..~~~~~~:~~~~~~....~~ o:~. ~+:..:~  ~~.   ~:~.~:::.                      |
|          ..   ....~~~~~~~~:~++.   :+~.~~ ~+~~.~ ~+:++++~                       |
| .         .             .~~~:::~..+~.... .~  .~.~~.~~                          |
| ...                          ..~~~.   .  ..~~~...                              |
|                         ....      ..  .~~:~~~...                               |
|~~~~~~...       .   . ..~::~~~~~~~~~~~.~..  ..~~~~~...                          |
|..~+++++:~:~~.~~~~~~~~~.~~:~:+::~::~+:~~~~         ......                       |
|..  .~~:~~::~~~:~~~::~:~::~~:+:~~.~~+:::~~..              . .                   |
|+:~~~~~~~~.~~~~~~~:~:::++:~++::~..~~..~~~.~~.. ..                               |
|~~.~~:~~.... . .  ..~..~~~~~~~~~.  ..      .   ...                              |
|:~:~:+:~~  .   .. ..               ..                                      MATS |
+--------------------------------------------------------------------------------+
```

File: cmd/rac/output.go [hunk 16]
```diff
@@ -0,0 +1,275 @@
...
+
+func infoSTAT() {
+	println(`
+### STAT.csv ###
+
+The following fields are read out exactly as they are encoded in the rac:
+SPID, SPREV, FPID, FPREV, SVNA, SVNB, SVNC, MODE, EDACE, EDACCE, EDACN,
+SPWEOP, SPWEEP, ANOMALY
+
+The following fields are omitted TS and TSS but instead replaced by:
```
Comment (skymandr) [inline cmd/rac/output.go]:
```suggestion
The fields TS and TSS are replaced by:

```

File: cmd/rac/output.go [hunk 17]
```diff
@@ -0,0 +1,276 @@
...
+
+func infoPM() {
+	println(`
+### PM.csv ###
+
+The following fields are read out exactly as they are encoded in the rac:
+PM1A, PM1ACNTR, PM1B, PM1BCNTR, PM1S, PM1SCNTR, PM2A, PM2ACNTR, PM2B,
+PM2BCNTR, PM2S, PM2SCNTR
+
+The following fields are omitted EXPTS and EXPTSS but instead replaced by:
```
Comment (skymandr) [inline cmd/rac/output.go]:
```suggestion
The fields EXPTS and EXPTSS are replaced by:
```

File: cmd/rac/output.go [hunk 18]
```diff
@@ -0,0 +1,276 @@
...
+| .         .             .~~~:::~..+~.... .~  .~.~~.~~                          |
+| ...                          ..~~~.   .  ..~~~...                              |
+|                         ....      ..  .~~:~~~...                               |
+|~~~~~~...       .   . ..~::~~~~~~~~~~~.~..  ..~~~~~...                          |
+|..~+++++:~:~~.~~~~~~~~~.~~:~:+::~::~+:~~~~         ......                       |
+|..  .~~:~~::~~~:~~~::~:~::~~:+:~~.~~+:::~~..              . .                   |
+|+:~~~~~~~~.~~~~~~~:~:::++:~++::~..~~..~~~.~~.. ..                               |
+|~~.~~:~~.... . .  ..~..~~~~~~~~~.  ..      .   ...                              |
+|:~:~:+:~~  .   .. ..               ..                                      MATS |
++--------------------------------------------------------------------------------+
```
Comment (local-minimum) [inline cmd/rac/output.go]:
```suggestion
 +--------------------------------------------------------------------------------+
 |..                 .                                             ..    .        |
 |    .                              .                                   ..       |
 |                                        .   .             ..            .       |
 |..                                                         .        . ~.        |
 | .                            ..    .             __.--´|`--.__         ..      |
 | .                                          __.--´|__.--|--._ |`--.__     .     |
 |                                      __.--´|__.--|--.__|__.--|--.__|`--.      .|
 |                                __.--´|__.--|--.__|__.--|--.__|__.--|_.-´   .   |
 |                          __.--´|__.--|--.__|__.--|--.__|__.--|_.--´            |
 |                         -.__.--|--.__|__.--´--.__|__.--|_.,-'*\                |
 |                         |   `--|__.--|--.__|__.--|_.--´  |*   |                |
 |..                       |         `--|__.--|_.--´    |   |.---´                |
 |o+o++:~~~..              |     __        '-´*         |   |*   `--.__           |
 |~::::++++++::~~~...      |    /  `--.     |#          |   .`-.__     `.         |
 |       ..~~~~~~:~~~~~~...|    |      \    |#    _.-\  |    |    `--._/          |
 |          ..   ....~~~~~~|    \      |#   |#   /   /* |    |`-._.-_.-\          |
 | .         .             |     `--.__/*   |#   \_.-*  |    |_.-´ /*  /          |
 | ...                      `--.__    *     |#          |_   |_--._\_.-           |
 |                         ....   `--.__    |*     __.--' `--´                    |
 |~~~~~~...       .   . ..~::~~~~~      `--.|__.--´~~~..                          |
 |..~+++++:~:~~.~~~~~~~~~.~~:~:+::~::~+:~~~~         ......                       |
 |..  .~~:~~::~~~:~~~::~:~::~~:+:~~.~~+:::~~..              . .                   |
 |+:~~~~~~~~.~~~~~~~:~:::++:~++::~..~~..~~~.~~.. ..             ~ .               |
 |~~.~~:~~.... . .  ..~..~~~~~~~~~.  ..      .   ...                 .            |
 |:~:~:+:~~  .   .. ..               ..                                   M A T S |
 +--------------------------------------------------------------------------------+
```

**Review by skymandr — APPROVED** [review]

**Pull request merged**